jicofo-common/src/main/java/org/jitsi/impl/protocol/xmpp/ChatRoomImpl.kt

```diff
@@ -0,0 +1,769 @@
+    override fun isMemberAllowedToUnmute(jid: Jid, mediaType: MediaType): Boolean =
+        avModeration.isAllowedToUnmute(mediaType, jid.toString())
+
+    // Use toList to avoid concurrent modification. TODO: add a removeAll to EventEmitter.
+    override fun removeAllListeners() = eventEmitter.eventHandlers.toList().forEach { eventEmitter.removeHandler(it) }
+
+    fun setStartMuted(startMuted: BooleanArray) = eventEmitter.fireEvent {
+        startMutedChanged(startMuted[0], startMuted[1])
+    }
+
+    @Throws(SmackException::class, XMPPException::class, InterruptedException::class)
+    override fun join() {
+        // TODO: clean-up the way we figure out what nickname to use.
+        resetState()
+        joinAs(xmppProvider.config.username)
+    }
+
+    /**
+     * Prepare this [ChatRoomImpl] for a call to [.joinAs], which send initial presence to
+     * the MUC. Resets any state that might have been set the previous time the MUC was joined.
+     */
+    private fun resetState() {
+        synchronized(members) {
+            if (members.isNotEmpty()) {
+                logger.warn("Removing " + members.size + " stale members.")
+                members.clear()
+            }
+        }
+        synchronized(this) {
+            role = null
+            lastPresenceSent = null
+            meetingId = null
+            logger.addContext("meeting_id", "")
+            isBreakoutRoom = false
+            mainRoom = null
+            avModeration.reset()
+        }
+    }
+
+    @Throws(SmackException::class, XMPPException::class, InterruptedException::class)
+    private fun joinAs(nickname: Resourcepart) {
+        myOccupantJid = JidCreate.entityFullFrom(roomJid, nickname)
+        presenceInterceptor = org.jivesoftware.smack.util.Consumer { presenceBuilder: PresenceBuilder ->
+            // The initial presence sent by smack contains an empty "x"
+            // extension. If this extension is included in a subsequent stanza,
+            // it indicates that the client lost its synchronization and causes
+            // the MUC service to re-send the presence of each occupant in the
+            // room.
+            synchronized(this@ChatRoomImpl) {
+                val p = presenceBuilder.build()
+                p.removeExtension(
+                    MUCInitialPresence.ELEMENT,
+                    MUCInitialPresence.NAMESPACE
+                )
+                lastPresenceSent = p.asBuilder()
+            }
+        }
+        if (muc!!.isJoined) {
+            muc.leave()
+        }
+        muc.addPresenceInterceptor(presenceInterceptor)
+        muc.createOrJoin(nickname)
+        val config = muc.configurationForm
+
+        // Read breakout rooms options
+        val isBreakoutRoomField = config.getField(MucConfigFields.IS_BREAKOUT_ROOM)
+        if (isBreakoutRoomField != null) {
+            isBreakoutRoom = java.lang.Boolean.parseBoolean(isBreakoutRoomField.firstValue)
```

Comment:
You shouldn't need the `java.lang` here.  Also it seems like this could be expressed in a more Kotlin-y way?